cardano-cli/src/Cardano/CLI/EraBased/Script/Spend/Read.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Cardano.CLI.EraBased.Script.Spend.Read
+  ( CliSpendScriptWitnessError
+  , readSpendScriptWitness
+  )
+where
+
+import           Cardano.Api
+import           Cardano.Api.Shelley
+
+import           Cardano.CLI.EraBased.Script.Mint.Types (CliScriptWitnessError (..))
+import           Cardano.CLI.EraBased.Script.Spend.Types
+import           Cardano.CLI.Read
+
+data CliSpendScriptWitnessError
+  = CliScriptWitnessError CliScriptWitnessError
+  | CliSpendScriptWitnessDatumError ScriptDataError
+
+instance Error CliSpendScriptWitnessError where
+  prettyError = \case
+    CliScriptWitnessError e -> prettyError e
+    CliSpendScriptWitnessDatumError e -> renderScriptDataError e
+
+readSpendScriptWitness
+  :: MonadIOTransError (FileError CliSpendScriptWitnessError) t m
+  => ShelleyBasedEra era -> CliSpendScriptRequirements -> t m (SpendScriptWitness era)
+readSpendScriptWitness sbe spendScriptReq =
+  case spendScriptReq of
+    OnDiskSimpleOrPlutusScript (OnDiskSimpleCliArgs simpleFp) -> do
+      let sFp = unFile simpleFp
+      s <-
+        modifyError (fmap (CliScriptWitnessError . SimpleScriptWitnessDecodeError)) $
+          readFileSimpleScript sFp
+      case s of
```

Comment:
```suggestion
      modifyError (fmap (CliScriptWitnessError . SimpleScriptWitnessDecodeError)) $
          readFileSimpleScript sFp
      >>= \case
```
I prefer to avoid temporary variables - but it's a matter of taste 😉

Reply:
I'll leave this one as is. It's strange to have an operator at the start of a newline.